src/main/java/com/github/fge/jsonschema/format/common/DateTimeAttribute.java

```diff
@@ -40,30 +40,29 @@
     extends AbstractFormatAttribute
 {
     private static final ImmutableList<String> FORMATS = ImmutableList.of(
-        "yyyy-MM-dd'T'HH:mm:ssZ", "yyyy-MM-dd'T'HH:mm:ss.[0-9]{1,12}Z"
+        "yyyy-MM-dd'T'HH:mm:ssZ", "yyyy-MM-dd'T'HH:mm:ss.[0-9]{1,9}Z"
     );
     private static final DateTimeFormatter FORMATTER;
 
     static {
-        final DateTimeParser secFracsParser = new DateTimeFormatterBuilder()
-            .appendLiteral('.').appendFractionOfSecond(1,12)
-            .toParser();
+        final DateTimeFormatter secFracsParser = new DateTimeFormatterBuilder()
+                .appendFraction(ChronoField.OFFSET_SECONDS, 1, 9, true)
+                .toFormatter();
 
-        DateTimeFormatterBuilder builder = new DateTimeFormatterBuilder();
-
-        builder = builder.appendFixedDecimal(year(), 4)
-            .appendLiteral('-')
-            .appendFixedDecimal(monthOfYear(), 2)
-            .appendLiteral('-')
-            .appendFixedDecimal(dayOfMonth(), 2)
-            .appendLiteral('T')
-            .appendFixedDecimal(hourOfDay(), 2)
-            .appendLiteral(':')
-            .appendFixedDecimal(minuteOfHour(), 2)
-            .appendLiteral(':')
-            .appendFixedDecimal(secondOfMinute(), 2)
-            .appendOptional(secFracsParser)
-            .appendTimeZoneOffset("Z", false, 2, 2);
+        DateTimeFormatterBuilder builder = new DateTimeFormatterBuilder()
+                .appendValue(ChronoField.YEAR, 4)
+                .appendLiteral('-')
+                .appendValue(ChronoField.MONTH_OF_YEAR, 2)
+                .appendLiteral('-')
+                .appendValue(ChronoField.DAY_OF_MONTH, 2)
+                .appendLiteral('T')
+                .appendValue(ChronoField.HOUR_OF_DAY, 2)
+                .appendLiteral(':')
+                .appendValue(ChronoField.MINUTE_OF_HOUR, 2)
+                .appendLiteral(':')
+                .appendValue(ChronoField.SECOND_OF_MINUTE, 2)
+                .appendOptional(secFracsParser)
+                .appendZoneOrOffsetId();
```

Comment:
Could be the cause of (overly?) strict offset validation, causing +0100 to be rejected.